### DISCUSSION TITLE
To deal with unrequited love

### DISCUSSION THREAD

**Discussion by ipaleka** [asastats/channel#583] [General]
As we don't want to take a winrar position in the ecosystem (ask @kerrilija what that means), I suppose it's time to break loose from this strange relationship we have established with Tinyman. At the end of this May it will be a total of 69 weeks that we allocate rewards to our users who provide liquidity to Tinyman, while they didn't have the courtesy for something Milesmile [can explain much better](https://discord.com/channels/906917846754418770/914540797054025788/1080272699198292080).

Here are the requirements for a Tinyman farm spot (https://discord.com/channels/906917846754418770/973245314687434832/1097936955351760907):

> by one of tinies:
> > the criteria for our selection is laid out in the blog: 
> > -A minimum TVL of $20,000 USD
> > -A trading volume of $100,000 USD in 2023
> > -High number and distributed shares of LPs in the pools

### Pact

As you may be aware, Pact has got farming programs too. You can access its UI at https://app.pact.fi/farms.

The optimal moment to switch to Pact is at the end of Governance period #7. At that time we'd be able to move all of our liquidity (Team's and prominent users') from Tinyman to Pact, at the same time securing that Pact's ASASTATS-ALGO LP tokens are eligible for staking in Governance period #8. Nevertheless, if an arrangement suits us, **we are ready to abandon the Tinyman pool immediately** - in that case, we'd "manually" compensate our users' loss for the Governance period #7.

I don't know if that counts, but we haven't used any AENEAS funds for the Pact pools. For those who will "negotiate" with the Pact guys, confirm our willingness to provide rewards in ASASTATS for the next 9-12 months, the farm will get a link on all the website pages, etc.

Feel free to add your suggestions!

**Comment by moloch10** [comment]
I'm down to move wherever so long as it's not in between governance periods.

**Comment by Dunohim** [comment]
Historical issues with Tinyman have clouded my confidence with the platform for quite some time already. That being said, they seem to have been on the ball more recently. (I'm yet to read the link from Milesmile)

While I have never been a direct user of the services that PACT offer, I've only ever seen good user reviews online. 

I'm here to state that I have no objections moving my LP position in line with the general consensus of the ASASTATS community and prominent (active) members.

**Comment by MochaN3rd** [comment]
I really like the UI/UX of Pact Fi, I'd prefer to wait until the end of Governance period #7 but if we're being reimbursed our rewards we would have lost, I'm fine with it...but wouldn't there need to be a snapshot first?

**Reply by ipaleka** [reply]
I'm not sure what do you mean by that "need for a snapshot"?

**Reply by MochaN3rd** [reply]
A snapshot for the users that were committed to Governance the whole time until the new farm launches? For example if I saw this discussion and pulled LP today, would I would still be eligible for  Gov reward reimbursement for the entire period? Imo that's not fair

**Reply by ipaleka** [reply]
Oh, I see, that's what I said by 'we'd "manually" compensate our users' loss for the Governance period', tho I don't think we should switch until `#7 ends...

Btw , in this case "snapshot" is just a word that better explains, actually we don't need to create a snapshot at that time, we just power up some script (any time in the future) that checks the status.

**Reply by moloch10** [reply]
I'd like to add that it's not just the ALGO payout to consider, Pera's NFTs are also part of the equation.

For that reason, I'm against moving mid cycle.

**Comment by AlgoRhythMatic** [comment]
I didn't realize Pact's TVL is ~1M > than total TVL of Tinyman, so a move seems reasonable from that market cap POV.  I am guessing a lack of rewards for remaining Tinyman liquidity will convince it all to eventually migrate so that there is no fragmentation of liquidity.  Do we know what the cost to Farm is at Pact?

Edit: maybe I misunderstood the function of Pact farm in our use case - would we still have an internal rewards program?

**Reply by ipaleka** [reply]
>  Do we know what the cost to Farm is at Pact?

I haven't started researching yet, also noone in Discord has replied with meaningful info so far.

> Edit: maybe I misunderstood the function of Pact farm in our use case - would we still have an internal rewards program?

Nope, we aim to move everything to Pact.

My wishful thinking is that still exists some kind of AENEAS program so we can switch for the next 9-12 months to rewards in ASASTATS we supply and eventual provider's rewards in ALGO...

**Reply by ipaleka** [reply]
Before we send any detail to anyone we find there in Pact, here's some data we probably have to provide in our communication with them. Feel free to add anything you think we should. In short, we aim for a 9 or 12 months long farm.

- - - - -

About:
ASA Stats develops, administrates and maintains all the aspects of the ASA Stats project until the ASA Stats DAO takes over the project.

ASA Stats website at https://www.asastats.com aims to bring price information for every ASA and NFT on Algorand Mainnet and it was the very first portfolio tracker software for the Algorand blockchain, published in November, 2022.

Since then, on top of implementing and deploying all the dApps deployed to Algorand Mainnet, ASA Stats API has been developed and it has been used by the ASA Stats Mobile app for iOS and Android, also developed by ASA Stats.

Investments:
US$23,800

Whitepaper:
https://www.asastats.com/whitepaper.pdf

Roadmap:
https://github.com/asastats/docs/blob/main/roadmap.md

Utilization:
https://github.com/asastats/channel/wiki/ASA-Stats-Token-utilization

Monthly transparency reports:
https://www.asastats.com/tokenomics/

Official communication channel:
https://github.com/asastats/channel

ASA Stats community channels:
https://discord.gg/Vjx7w7pAC7
https://twitter.com/asastatscom
https://www.reddit.com/r/asastats/

[ASA Stats Token staking and farming historic data](https://github.com/asastats/channel/discussions/448#discussioncomment-4810812)

**Reply by AlgoRhythMatic** [reply]
> Nope, we aim to move everything to Pact.

> My wishful thinking is that still exists some kind of AENEAS program so we can switch for the next 9-12 months to rewards in ASASTATS we supply and eventual provider's rewards in ALGO...

The reason I asked: it looks like all Pact farms earn rewards in ALGO -or- ALGO + farmed coin.  I didn't see any that only rewarded w/ the farmed coin.  I wonder if these are Aeneas rewards in play?  

It also looks like @MochaN3rd also just posted in Discord about someone named Choppa from Pact, so hoping he will comment here as well!

**Comment by AlgoRhythMatic** [comment]
Wondering if we still have thoughts to migrate liquidity to Pact?  I'm still not clear about how their farms work (who provides the ALGO portion of the rewards).

**Reply by ipaleka** [reply]
> Wondering if we still have thoughts to migrate liquidity to Pact? I'm still not clear about how their farms work (who provides the ALGO portion of the rewards).

I do, tho the reasoning why is unclear besides moving away from Tinyman ... The thing is that there is no ALGO portion of the rewards.

**Reply by AlgoRhythMatic** [reply]
Assuming Gard confirm there is no grant-related ALGO potential - one option could be to identify the allocation of rewards in ASASTATS, convert half to ALGO, and then provide them as 50/50 split.  This is honestly what I do now each week when I receive my liquidity rewards and re-invest in the pool.

**Reply by ipaleka** [reply]
Our primary problem has been funding the development and for the last 17 months we have avoided selling any ASASTATS for that cause. 

Selling ASASTATS to reward liquidity providers implies secured development for *years* and we are still far from there.

**Reply by ipaleka** [reply]
> Assuming Gard confirm there is no grant-related ALGO potential 

You mean Pact, right? I sent a Discord DM to Choppa from Pact earlier today:

- - - - -
Hi again! 🙂
I'm just reading governance page info:
>The Algorand Foundation recently piloted a deployment of 42MM ALGO of treasury liquidity to Folks Finance, Pact, AlgoRai Finance, and Tinyman with the objective of contributing to ecosystem liquidity, supporting TVL growth, and strengthening the DApps’ critical mass.

May you tell me what has been a condition to get some of those ALGO for our Pact ASASTATS-ALGO liquidity? 
I mean, that's something we definitely search for to switch to Pact. 
- - - - -

**Reply by AlgoRhythMatic** [reply]
> You mean Pact, right?

Yes!

**Comment by ipaleka** [comment]
I've got a chat in Discord DM with Choppa from Pact for the last week(s) and we'd probably qualify for the rewards in ALGO for the governance period 8.

Based on our current liquidity, we may start with the existing pool https://app.pact.fi/pools?page=1&search=699569010#table having a fee of 0.3%, or we may create a new one having a higher fee.

As of today, our plan is that on Mon 19 and Mon 26, we send the two last allocations from our in-house incentive program with a note that we move our incentive to Pact pool and farm. The plan, as of now, is that we keep that farm for at least 9 or 12 weeks.

We need a final discussion where we should define which pool we should use (the existing or a new one) and how many ASASTATS we should allocate for rewards. The plan is to set the rewards for the first week and then just adjust them for week 2 which should stay for months or so...

**Reply by ipaleka** [reply]
https://discord.com/channels/906917846754418770/914540797054025788/1120317865665507349

-----
So here's initial draft for the today's rewards allocation notes:
```We're moving to Pact farming starting July 1st and Governance period #8! Please remove your liquidity from Tinyman and add it to https://app.pact.fi/add-liquidity/1129173571 ; ASA Stats Official LP rewards 2023-06-18 2.0: ID: xy```
https://app.pact.fi/add-liquidity/1129173571

**Reply by ipaleka** [reply]
motuwagon — Today at 3:39 PM
So best time to change over after next week's rewards or no time is just best before July 1st.

-----
ipaleka — Today at 3:53 PM
I guess I miscalculated the week our in-house incentive program will end, it's week 73.

So, removing liquidity from Tinyman and adding it to Pact **on Sunday, 2nd of July** is completely safe from both perspectives: week 73 rewards of our incentive program will be counted against Sunday 2nd at 00:00, while governance period #7 rewards are safe even a day longer.
